File: leetcode/167 Two Sum II - Input Array Is Sorted/solution.go
func twoSum(numbers []int, target int) []int {
    var n int = len(numbers)
    p1 := 0; // 最简单的双指针练习题！！！
    p2 := n-1;
    result := make([]int,2)
    for i:=0; i<n; i++ {
        sum := numbers[p1] + numbers[p2]
        if sum == target {
            result[0], result[1] = p1+1, p2+1
        } else if sum < target {
            p1++
        } else {
            p2--
        }
    }
    return result
}

File: leetcode/167 Two Sum II - Input Array Is Sorted/solution.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int n = numbers.length;
        int p1 = 0; // 最简单的双指针练习题！！！
        int p2 = n-1;
        int[] result = new int[2];
        while (p1 < p2) {
            int sum = numbers[p1] + numbers[p2];
            if (sum == target) {
                result[0] = p1+1;
                result[1] = p2+1;
                return result;
            } else if (sum < target) {
                p1++;
            } else {
                p2--;
            }
        }
        return result;
    }
}